Consul/Catalog.cs

```diff
@@ -118,6 +118,62 @@ public class ServiceAddress
         public int Port { get; set; }
     }
 
+    public class CompoundServiceName
+    {
+        public string Namespace { get; set; }
+        public string Partition { get; set; }
+        public string Name { get; set; }
+    }
+
+    public class GatewayService
+    {
+        public CompoundServiceName Gateway { get; set; }
+        public CompoundServiceName Service { get; set; }
+        public ServiceKind GatewayKind { get; set; }
+        public int Port { get; set; }
+        public string Protocol { get; set; }
+        public List<string> Hosts { get; set; }
+        public string CAFile { get; set; }
+        public string CertFile { get; set; }
+        public string KeyFile { get; set; }
+        public string SNI { get; set; }
+        public bool FromWildcard { get; set; }
+    }
+
+    public class TerminatingGatewayConfigEntry : IConfigurationEntry
```

Comment:
We already have a `TerminalGatewayEntry` in the `Configuration.cs` file so there is no need to re-define it.
I think the `TerminalGatewayEntry` and associated `LinkedService` are incorrectly defined, but feel free to fix them.
It will break backward compatibility, but those structs are likely not used by anyone since they are not correctly defined.

Reply:
Oh, I am surprised how I missed it. Some of the classes were a bit incorrect. TerminatingGatewayEntry had a wrong name, LinkedService has wrong class structure, TerminatingGatewayEntry had excess fields. IngressGatewayEntry wasn't implementing IConfigEntry. Also I could rename them to TerminatingGatewayConfigEntry and IngressGatewayConfigEntry as in Golang client code, but that isn't critical.